File: pfproject_py/src/pfgame.py
# !/usr/bin/env python3
# -*- coding:utf-8 -*-
__author__ = 'Zhiquan Wang'
__date__ = '2018/7/20 22:30'

from pfmap import *
from pfrule import *
from pfplayer import *
import time
import md5manager


class PixelFightGame(object):

    def __init__(self):
        self.__game_rule = PixelFightRule()
        self.__round_counter = 0
        self.__player_info_list = []
        self.__pixel_map = PixelMap(self.__game_rule.map_height, self.__game_rule.map_width)

    def launch_game(self):
        pass

    def load_game_rule(self):
        pass

    def gen_player_id(self, _uname, _peer_info):
        print('in')
        print(_uname)
        print(_peer_info)
        tmp_id = md5manager.create_md5((_uname, _peer_info, time.time()))
        print(tmp_id)
        tmp_p = PixelFightPlayer(usr_name=_uname, login_id=tmp_id)
        self.__add_player(tmp_p)
        return tmp_id

    def __add_player(self, _p):
        self.__player_info_list.append(_p)

    def attack_grid(self, _x, _y, _player_info):
        pass

File: pfProject_CS/TrialProject/TrialProject/src/PfGame.cs
using System;

namespace TrialProject.src
{
    public class PixelFightGame
    {
        private string gameRule;
        private string playerInfoList;
        private string pixelFightMap;
        private string perPosList;
        private string visionManager;
        private bool isPause;
        private bool isReady;
        private int gameRatio;
        private int roundCounter;
        private int _mw;
        private int _mh;
        public PixelFightGame()
        {
            GameRule = pixelFightRule(_mw = 30, _mh = 30);
            RoundCounter = 0;
            GameRatio = 0;
            PlayerInfoList = null;
            PixelFightMap = pixelMap();
            PerPosList = null;
            VisionManager = null;
            IsPause = false;
            IsReady = false;
        }
        private string pixelMap()
        {
            throw new NotImplementedException();
        }
        private string pixelFightRule(int _mw, int _mh)
        {
            throw new NotImplementedException();
        }

        public string GameRule { get => gameRule; set => gameRule = value; }
        public string PlayerInfoList { get => playerInfoList; set => playerInfoList = value; }
        public string PixelFightMap { get => pixelFightMap; set => pixelFightMap = value; }
        public string PerPosList { get => perPosList; set => perPosList = value; }
        public string VisionManager { get => visionManager; set => visionManager = value; }
        public bool IsPause { get => isPause; set => isPause = value; }
        public bool IsReady { get => isReady; set => isReady = value; }
        public int RoundCounter { get => roundCounter; set => roundCounter = value; }
        public int GameRatio { get => gameRatio; set => gameRatio = value; }

        
    }
}